src/App.js

```diff
@@ -1,47 +1,64 @@
 /* src/App.js */
-import React from 'react';
+import React, { useState } from 'react';
 import Amplify from 'aws-amplify';
 import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
 import { withAuthenticator, AmplifySignOut } from '@aws-amplify/ui-react';
 
 import awsExports from './aws-exports';
 import Quiz from './components/Quiz';
 import CocktailForm from './components/Form';
+import Loader from './components/Loader';
 
 Amplify.configure(awsExports);
 
-function App() {
+const App = () => {
+	const [displayLoader, changeLoader] = useState(false);
+
+	const handleLoaderUpdate = (display, message) => {
+		changeLoader(display, message);
+	};
+
+	const showLoader = () => {
+		if (displayLoader) {
+			return <Loader />;
+		}
+	};
+
 	return (
-		<Router>
-			<div className='App'>
-				<nav className='header-nav'>
-					<ul>
-						<li>
-							<Link to='/quiz'>Cocktail Quiz</Link>
-						</li>
-						<li>
-							<Link to='/cocktailform'>Add Cocktails</Link>
-						</li>
-						<li className='amp-sign-out-button'>
-							<AmplifySignOut />
-						</li>
-					</ul>
-				</nav>
-
-				<Switch>
-					<Route path='/quiz'>
-						<Quiz />
-					</Route>
-					<Route path='/cocktailform'>
-						<CocktailForm />
-					</Route>
-					<Route path='/'>
-						<Quiz />
-					</Route>
-				</Switch>
-			</div>
-		</Router>
+		<div>
+			<Router>
+				<div className='App'>
+					<nav className='header-nav'>
+						<ul>
+							<li>
+								<Link to='/quiz'>Cocktail Quiz</Link>
+							</li>
+							<li>
+								<Link to='/cocktailform'>Add Cocktails</Link>
+							</li>
+							<li className='amp-sign-out-button'>
+								<AmplifySignOut />
+							</li>
+						</ul>
+					</nav>
+
+					<Switch>
+						<Route path='/quiz'>
+							<Quiz onLoaderUpdate={handleLoaderUpdate} />
+						</Route>
+						<Route path='/cocktailform'>
+							<CocktailForm onLoaderUpdate={handleLoaderUpdate} />
+						</Route>
+						<Route path='/'>
+							<Quiz />
+						</Route>
+					</Switch>
+				</div>
+			</Router>
+
+			{showLoader()}
```

Comment:
instead of using that if in the function above, you can just do `{displayLoader && <Loader />}`.  This is called a short circuit in React - https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator